src/data/AmazonReviewNew.py

```diff
@@ -0,0 +1,139 @@
+# -*- coding: utf-8 -*-
+from pathlib import Path
+from typing import Optional
+from os.path import exists
+
+from torch.utils.data import DataLoader
+from torch.utils.data.sampler import SubsetRandomSampler
+from torchtext.datasets import AmazonReviewPolarity
+import pytorch_lightning as pl
+import torch
+import numpy as np
+
+
+project_dir = Path(__file__).resolve().parents[2]
+
+
+class AmazonReviewFullDataModule(pl.LightningDataModule):
+    def __init__(
+        self,
+        tokenizer,
+        val_size: float = 0.2,
+        data_dir: str = project_dir.joinpath("data"),
+        batch_size: int = 32,
+        max_seq_length: int = 128,
+    ):
+        super().__init__()
+        self.data_dir = data_dir
+        self.batch_size = batch_size
+        self.val_size = val_size
+        self.max_seq_length = max_seq_length
+
+        self.tokenizer = tokenizer
+
+        self.processed_data_path = "data/processed/"
+
+        train_path = "data/processed/train.pt"
+        test_path = "data/processed/test.pt"
+        if not exists(train_path) or not exists(test_path):
+            self.setup()
+
+        # self.tokenized_train = torch.load(train_path)
+        self.tokenized_test = torch.load(test_path)
+
+        # self.train_sampler,self.valid_sampler = self.get_train_val_samplers()
+
+    def get_train_val_samplers(self):
+        n = len(self.tokenized_train)
+        indices = list(range(n))
+        valid_size = 0.20
```

Comment:
Best practice would be to have this as a class attribute 

```suggestion
        valid_size = self.valid_size
````